app/src/main/java/com/example/voyagerx/ui/fragments/landing/list/LaunchOverviewAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.voyagerx.ui.fragments.landing.list
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.voyagerx.databinding.LandingPageOverviewCardBinding
+import com.example.voyagerx.helpers.LaunchClickListener
+import com.example.voyagerx.repository.model.Launch
+
+class LaunchOverviewAdapter(private val listener: LaunchClickListener) :
+    RecyclerView.Adapter<LaunchOverviewViewHolder>() {
+    private var launches: List<Launch> = listOf()
+
+    fun initializeList(launches: List<Launch>) {
+        this.launches = launches
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LaunchOverviewViewHolder =
+        LaunchOverviewViewHolder(
+            LandingPageOverviewCardBinding.inflate(
+                LayoutInflater.from(parent.context), parent, false
+            )
+        )
+
+    override fun onBindViewHolder(holder: LaunchOverviewViewHolder, position: Int) {
+        val launch = launches[position]
+        holder.bind(launch)
+        holder.itemView.setOnClickListener {
+            listener.onClick(launch)
```

Comment:
How does listener have an `onClick` method?

Reply:
Good question...this was copy-pasted but I will try to explain. I basically set up the class `LaunchClickListener` that takes in a lambda function during construction and calls that lambda within its onClick method. The listener that you highlighted is constructed inside `LandingPageFragment` within `setupList`. So in my case that onClick will call a lambda that logs the launch data inside the launch view holder.